api/types/types.go

```diff
@@ -212,7 +213,10 @@ type Info struct {
 	SecurityOptions     []string
 	ProductLicense      string               `json:",omitempty"`
 	DefaultAddressPools []NetworkAddressPool `json:",omitempty"`
-	Warnings            []string
+	// RootlessKit is unmarshallable into a struct of github.com/rootless-containers/rootlesskit/pkg/api.Info .
+	// (not unmarshalled by default, so as to propagate all the fields that are unknown to github.com/rootless-containers/rootlesskit/pkg/api.Info .)
+	RootlessKit json.RawMessage `json:",omitempty"`
```

Comment:
I'm a bit on the fence on adding this as a new field. We originally added the `ContainerdCommit`, `RuncCommit` and `InitCommit` fields when those versions _had to_ match a specific version/commit (API's and their behavior wasn't really  "stable" yet, and we received various bugreports that turned out to be because non-default versions were installed).

The `RuncCommit` already has shown to be a bit problematic for situations where another OCI runtime is installed (`crun`, e.g.). We fixed those, but it's still a bit odd to have hard-coded `Runc` now containing "non-runc" information.

In short; I would like to sunset those fields, and instead consider using the `Components` information that we currently use in the `/version` endpoint; https://github.com/moby/moby/blob/7b9275c0da707b030e62c96b679a976f31f929d3/daemon/info.go#L98-L114


I _think_ most (if not all) of this information is for "informational" purposes, and not a strict "API" contract , so having the information (other than `Version` and `Name`) in a more "free-form" format (just `key/value` information) would probably work, and doing so would keep the API more "stable" (and flexible)

WDYT?

Reply:
The purpose of this field is to show network driver and port driver info rather than the version, so types.ComponentVersion doesn't seem to fit here IMO.

Reply:
The `types.ComponentVersion` allows to provide arbitrary other information. The only limitation is that it has a single level key/value pairs, not nested, so relevant information that's nested would have to be slightly changed, but probably would allow for better formatting when presenting in `docker info`, e.g. something like:

```go
Components: []types.ComponentVersion{
    {
        Name:    "rootlesskit",
        Version: "v0.14.2",
        Details: map[string]string{
            "apiVersion":        "1.1.1",
            "stateDir":          "/tmp/rootlesskit648249121",
            "networkDriver":     "slirp4netns",
            "networkDriver.dns": "10.0.2.3, 8.8.8.8",
            "portDriver":        "builtin",
            "portDriver.protos": "tcp, tcp4, tcp6, udp, udp4, udp6",
        },
    },
},
```

This is similar to what we do for storage drivers (only that one uses a slice of `key value` pairs;

<img width="626" alt="Screenshot 2021-05-01 at 14 01 42" src="https://user-images.githubusercontent.com/1804568/116781954-9a89e500-aa86-11eb-8ed1-2a64a583bf49.png">
<img width="331" alt="Screenshot 2021-05-01 at 14 01 50" src="https://user-images.githubusercontent.com/1804568/116781956-9c53a880-aa86-11eb-8262-405c8529fc67.png">

Reply:
updated to add rootlesskit version to `/version` endpoint.

Adding `Components`  to `/info` would be another PR.